readme.md

```diff
@@ -59,12 +52,98 @@ class Article extends Model
     ...
 ```
 
-## Customization
+---
+
+If you intend to display the activity log shortcut button on CRUD List, enable each button using the following approach:
+
+```diff
+<?php
+
+namespace App\Http\Controllers\Admin;
+
+...
+
+class ArticleCrudController extends CrudController
+{
+    ...
++    use \Backpack\ActivityLog\Http\Controllers\Operations\ShowModelActivityLogsOperation;
++    use \Backpack\ActivityLog\Http\Controllers\Operations\ShowEntryActivityLogsOperation;
+    ...
+```
+
+**ShowModelActivityLogsOperation**: Displays a button at the Model level, allowing users to access a comprehensive list of all Model activities.  
+**ShowEntryActivityLogsOperation**: Adds line buttons to each entry in the list, enabling users to view activity logs specific to individual entries.
+
+---
+
+This package also offers shortcuts for Causers, entities that initiate activities on other entities. While typically users, other entities can trigger changes too.  
+To enable the activity log shortcut button on causers:
+
+```diff
+<?php
+
+namespace App\Http\Controllers\Admin;
+
+...
+
+class ExampleCrudController extends CrudController
+{
+    ...
++    use \Backpack\ActivityLog\Http\Controllers\Operations\ShowCauserModelActivityLogsOperation;
++    use \Backpack\ActivityLog\Http\Controllers\Operations\ShowCauserEntryActivityLogsOperation;
+    ...
+```
+
+---
+
+**Quick note**: If you haven't customized your `UserCrudController` yet, you can achieve this by binding the backpack controller to your custom controller.
+
+`App\Providers\AppServiceProvider.php`
+```php
+class AppServiceProvider extends ServiceProvider
+{
+    public function register()
+    {
+        $this->app->bind(
+            \Backpack\PermissionManager\app\Http\Controllers\UserCrudController::class,
+            \App\Http\Controllers\Admin\UserCrudController::class
+        );
+```
+
+`App\Http\Controllers\Admin\UserCrudController.php`
+```php
+use Backpack\PermissionManager\app\Http\Controllers\UserCrudController as OriginalUserCrudController;
+
+class UserCrudController extends OriginalUserCrudController
+{
+    use \Backpack\ActivityLog\Http\Controllers\Operations\ShowCauserModelActivityLogsOperation;
+    use \Backpack\ActivityLog\Http\Controllers\Operations\ShowCauserEntryActivityLogsOperation;
+}
+```
+
+## FAQ
+
+#### What gets logged by default?
+
+By default, the backpack activity logger records changes to the **fillable fields** that have been **modified**.
+
+#### How do you customize what gets logged?
+
+To customize the logged information, you can override the `getActivitylogOptions` method in your models.
+
+```php
+public function getActivitylogOptions(): LogOptions
+{
+    return LogOptions::defaults()
+        ->logFillable()
+        ->logOnlyDirty();
+}
+```
```

Comment:
Maybe we need to rethink this. Previously, we thought of making the package plug-and-play. Install it and that's it. But it looks like **we can't actually make it plug-and-play**. Devs will still need to make all of their models use a `LogActivity` trait.

So let's keep in mind the use cases for this package:
- 60% - a developer who needs to start tracking admin activity is installing this, and wants some or all of their Models tracked;
- 20% - a developer who is already using `spatie/activity-log` wants an interface for it; they are already logging stuff - maybe model changes, maybe custom activities;
- 5% - a developer who is unhappy with Revisionable is installing this, as an alternative;

---

What I'm saying is... maybe it's best to do as little custom stuff in the Model as we can, and let the developer use the Spatie package. Then... we only provide an interface for that package.

We are unhappy with Spatie's default logging. I know. Personally I would have loved to have an option to just enable logging for all models, all attributes, only dirty. Without having to add the trait. But... it is what it is. I think it's easier for us to maintain... and for the developer to understand... if we just instruct them to add the Spatie trait and Spatie needed function to their models. Our recommandation is to `LogOptions::defaults()->logAll()->logOnlyDirty();` but if they want to do something else... fine... our interface will still work and be useful.

Reply:
I'm suggesting `logAll()` instead of `logFillable()` because some models might be using guarded, some might be using fillable... and `logAll()` covers everything. As long as we also put `logOnlyDirty()` the entries will not get that big.

Reply:
I've re-evaluated this today and... if the only thing we do in our trait is to use some Spatie method for defaults... I think we should delete our trait, and instruct them to use Spatie's. That way is the cleanest. That part really isn't part of this package, it's a part of Spatie's package.

Reply:
I mainly agree, but Spaties doesn't provide defaults. This method is providing a default that will cover +90 use cases, and will allow OUR package to be used by just adding a Trait to the models.
Without this, for every model developers will need the trait + this `getActivitylogOptions` definition 😕

Reply:
😔 you're right. And I know you are right. And I hate that you are right 😔

![sad-sponge-bob-gif-by-spongebob-squarepants](https://github.com/Laravel-Backpack/activity-log/assets/1032474/86ad51b9-ffac-4be2-8dd3-8c7859fef423)

Fine, we keep it. We can clarify in the docs that they should MANUALLY make their models ActivityLog-able OR they can just use the trait if they want our defaults.

Reply:
> What do you think about defaulting to logAll() instead of logFillable()? See any problems in doing that?

No ... since we're filtering later for the fillable, it's ok 🤷‍♂️